stat/spatial/spatial.go

```diff
@@ -43,11 +42,21 @@ func GetisOrdGStar(i int, data, weights []float64, locality mat.Matrix) float64
 	n := float64(len(data))
 	mean, std := stat.MeanStdDev(data, weights)
 	var dwd, dww, sw float64
-	for j, v := range data {
-		w := locality.At(i, j)
-		sw += w
-		dwd += w * v
-		dww += w * w
+	if band, ok := locality.(mat.Banded); ok {
```

Comment:
I have a solution. If we reduce the `Apply` method from its current signature to `Apply(func(i, j int, v float64))`, then an `Applier` interface can be used. This makes the `Apply` method slightly less efficient for its current use since `a` would need to be referred to in a closure, but I think the trade-off is probably worth it. Then we just add `Apply` to each of our types and note in the documentation that it only applies to the filled region of the receiver. Alternatively, a new `ApplyFilled` with the signature above could be invented.